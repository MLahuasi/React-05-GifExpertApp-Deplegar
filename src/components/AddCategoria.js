import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategoria = ({ setCategorias }) => {

    const [inputValue, setInputValue] = useState(''); //Si no se envia un parámetro en useState se presenta un error en consola ya que useState es undefined

    const handleInputChange = ( e ) => {
        setInputValue(e.target.value); //Escribe en la caja de texto
    };

    const handleSubmit = (e) => {
        e.preventDefault(); //Previene el refresh de la página

        if( inputValue.trim().length > 2 ){
            setCategorias(cats => [inputValue, ...cats ]);
            setInputValue('');
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <input
                type="text"
                value={inputValue}
                onChange={handleInputChange}
            />
        </form>
    )
}

AddCategoria.propTypes = {
    setCategorias: PropTypes.func.isRequired
}