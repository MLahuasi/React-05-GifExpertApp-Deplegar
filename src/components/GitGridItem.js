import React from 'react'

export const GitGridItem = ({id, title, url}) => {

    return (
        <div className="card animate__animated animate__bounce">
            <img src={url} alt={ title } />
            <p>{title}</p>
        </div>
    )
}
