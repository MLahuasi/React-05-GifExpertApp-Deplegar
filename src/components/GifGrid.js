import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GitGridItem } from './GitGridItem';

export const GifGrid = ({ categoria }) => {

    /**
     * Destructuraci√≥n al llamar useFetchGifs
     */
    const { data:images, loading } = useFetchGifs(categoria);
    return (
        <>
            <h3 className='animate__animated animate__fadeIn'>{categoria}</h3>

            {loading && <p className='animate__animated animate__flash'>Loading</p> }

            <div className="card-grid">
                {
                    images.map( img => (
                        <GitGridItem
                            key={img.id}
                            { ...img}
                        />
                    ))
                }
            </div>

        </>
    )
}

