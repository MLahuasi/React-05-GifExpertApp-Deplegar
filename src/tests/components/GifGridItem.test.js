import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import { GitGridItem }  from '../../components/GitGridItem';

describe('Test GitGridItem.js', () => {


    let id = 'aAbax5anloMNk6TSP9';
    let title = 'Dragon Ball GIF by TOEI Animation UK';
    let url = 'https://media4.giphy.com/media/aAbax5anloMNk6TSP9/giphy.gif?cid=85b515a5vue5x4b3l0v8zlb9ilfhqx9bx707skt72x36dyn5&rid=giphy.gif&ct=g';

    let obj = {
        id,
        title,
        url
    };

    let wrapper = shallow(<GitGridItem id={id} title={title} url={url} />);
    //Es igual a:
    //let wrapper = shallow(<GitGridItem {...obj} />);

    test('Debe mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe tener un parráfo con el título', () => {
        const p = wrapper.find('p');  //Busca Parrafo
        expect( p.text().trim()).toBe( title ); //Compara texto de párrafo con el título enviado
    });

    test('Debe el src del imagen igual al url que se envía como parámetro, de igual manera con alt', () => {
        const img = wrapper.find('img');
        //console.log( img.html() );  //Retorna el html de image

        //console.log( img.props() );   //Retorna las propiedades de image, por ejemplo src y alt

        const { src, alt } = img.props();
        // console.log('src', src);
        // console.log('alt', alt);

        expect( src ).toBe( url );
        expect( alt ).toBe( title );
    })

    test('Debe tener la clase css animate__bounce', () => {
        const div = wrapper.find('div');
        const { className } = div.props();
        console.log(className);

        expect(className).not.toMatch(/ animate__fadeIn/);
        expect(className).toMatch(/ animate__bounce/);      //ToMatch valida si un string contiene una expresión regular
        //Ejemplo Clase
        expect(className.includes('animate__bounce')).toBe(true);

    })

});




