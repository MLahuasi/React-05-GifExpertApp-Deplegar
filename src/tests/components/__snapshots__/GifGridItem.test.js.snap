// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GitGridItem.js Debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="Dragon Ball GIF by TOEI Animation UK"
    src="https://media4.giphy.com/media/aAbax5anloMNk6TSP9/giphy.gif?cid=85b515a5vue5x4b3l0v8zlb9ilfhqx9bx707skt72x36dyn5&rid=giphy.gif&ct=g"
  />
  <p>
    Dragon Ball GIF by TOEI Animation UK
  </p>
</div>
`;
